src/main/java/org/veupathdb/service/sr/service/SequenceRetrievalAsyncService.java

```diff
@@ -54,60 +48,71 @@ private JobResponse asyncResponse(
+    spec.setBasesPerLine(basesPerLine);
+    spec.setSequenceType(sequenceType);
+    return PostSequencesAsyncBySequenceTypeResponse.respond200WithApplicationJson(asyncResponse(spec));
   }
 
   @Override
   public PostSequencesAsyncBySequenceTypeAndFileFormatResponse postSequencesAsyncBySequenceTypeAndFileFormat(
-      String sequenceTypeStr,
+      String sequenceType,
       String fileFormatStr,
       DeflineFormat deflineFormat,
-      Boolean forceStrandedness,
       Integer basesPerLine,
       StartOffset startOffset,
       SequencesAsyncSequenceTypeFileFormatPostMultipartFormData entity){
 
-      var uploadMethod = entity.getUploadMethod();
-      var file = entity.getFile();
-      var url = entity.getUrl();
+    var file = entity.getFile();
+    var url = entity.getUrl();
 
-    // throw not found since these are path params
-    var fileFormat = EnumUtil.validate(fileFormatStr, SupportedFileFormat.values(), NotFoundException::new);
-    var sequenceType = EnumUtil.validate(sequenceTypeStr, SequenceType.values(), NotFoundException::new);
-
-    try (InputStream fileStream = switch (uploadMethod) {
-      case FILE -> new FileInputStream(file);
-      case URL -> new URL(url).openStream();
-    }) {
-
-      var requestedFeatures = switch (fileFormat) {
-        case BED -> ConvertFeatures.featuresFromBed(fileStream, startOffset);
-        case GFF3 -> ConvertFeatures.featuresFromGff3(fileStream);
-      };
-
-      var spec = Sequences.getSequenceSpec(sequenceType);
-      var index = Sequences.getSequenceIndex(sequenceType);
-      var sequences = Sequences.getSequenceFile(sequenceType);
+    var spec = new SequenceRetrievalSpecImpl();
 
-      var features = Validate.getValidatedFeatures(sequenceType, index, requestedFeatures, forceStrandedness, spec);
+    spec.setDeflineFormat(deflineFormat);
+    spec.setBasesPerLine(basesPerLine);
+    var fileFormat = EnumUtil.validate(fileFormatStr, SupportedFileFormat.values(), NotFoundException::new);
+    spec.setFileFormat(fileFormat);
+    spec.setSequenceType(sequenceType);
 
-      return PostSequencesAsyncBySequenceTypeAndFileFormatResponse.respond200WithApplicationJson(asyncResponse(features, sequenceType, deflineFormat, forceStrandedness, basesPerLine));
+    var uploadMethod = entity.getUploadMethod();
+    switch(uploadMethod){
+      case FILE:
+        spec.setFeaturesStr(slurp(file));
+        break;
+      case URL:
+        spec.setFeaturesUrl(url);
+        break;
+    }
+    return PostSequencesAsyncBySequenceTypeAndFileFormatResponse.respond200WithApplicationJson(asyncResponse(spec));
+  }
 
-    } catch (IOException e) {
-      throw new RuntimeException("Unable to complete file processing", e);
+  private static String slurp(File file){
+    try {
+      return getFileContent(new FileInputStream(file), "UTF-8");
+    } catch (IOException e){
+      throw new RuntimeException(e);
     }
   }
+
+  //https://stackoverflow.com/a/15161590
```

Comment:
You don't need to write this code.  If you're not concerned about size and want to read everything into a String, this is simpler:
```
ByteArrayOutputStream out = new ByteArrayOutputStream();
in.transferTo(out);
return out.toString();
```